core/trino-spi/src/main/java/io/trino/spi/connector/Connector.java

```diff
@@ -147,6 +147,11 @@ default Set<TableProcedureMetadata> getTableProcedures()
         return emptySet();
     }
 
+    default long getExtraMemoryRequirement()
```

Comment:
I would call this "getInitialMemoryRequirement" to indicate that this is a memory allocation needed during initialization. 

Also, add javadocs explaining that this will get called once when the plugin is first instantiated, whether the allocation is per catalog or shared across instantiations, and any other relevant details.